File: 4138-minimum-deletion-cost-to-make-all-characters-equal/minimum-deletion-cost-to-make-all-characters-equal.js
/**
 * @param {string} s
 * @param {number[]} cost
 * @return {number}
 */
var minCost = function(s, cost) {
    let arr=new Array(26).fill(0);
    let total=0
    for(let i=0;i<cost.length;i++){
        total+=cost[i];
        arr[s.charCodeAt(i)-'a'.charCodeAt(0)]+=cost[i];
    }
    let max=0;
    for(let x of arr){
        max=Math.max(max,x);
    }
    return total-max;
};

File: 4138-minimum-deletion-cost-to-make-all-characters-equal/minimum-deletion-cost-to-make-all-characters-equal.java
class Solution {
    public long minCost(String s, int[] cost) {
        long total=0,max=0;
        long arr[]=new long[26];
        for(int i=0;i<s.length();i++){
            total+=cost[i];
            arr[s.charAt(i)-'a']+=cost[i];
            max=Math.max(max,arr[s.charAt(i)-'a']);
        }
        return total-max;
    }
}